src/query/query_hash_scan.c

```diff
@@ -0,0 +1,441 @@
+  while (regu_list != NULL)
+    {
+      rc = fetch_peek_dbval (thread_p, &regu_list->value, vd, NULL, NULL, NULL, &key->values[key->val_count]);
+      if (rc != NO_ERROR)
+	{
+	  return rc;
+	}
+
+      /* next */
+      regu_list = regu_list->next;
+      key->val_count++;
+    }
+
+  /* all ok */
+  return NO_ERROR;
+}
+
+/*
+ * qdata_print_hash_scan_entry () - Print the entry
+ *                              Will be used by mht_dump() function
+ *   return:
+ *   fp(in)     :
+ *   key(in)    :
+ *   data(in)   :
+ *   args(in)   :
+ */
+int
+qdata_print_hash_scan_entry (THREAD_ENTRY * thread_p, FILE * fp, const void *key, void *data, void *args)
+{
+  HASH_SCAN_VALUE *data2 = (HASH_SCAN_VALUE *) data;
+  HASH_SCAN_KEY *key2 = (HASH_SCAN_KEY *) key;
+  QFILE_TUPLE_VALUE_TYPE_LIST *list = (QFILE_TUPLE_VALUE_TYPE_LIST *) args;
+
+  if (!data2 || !key2)
+    {
+      return false;
+    }
+  if (fp == NULL)
+    {
+      fp = stdout;
+    }
+
+  fprintf (fp, "LIST_CACHE_ENTRY (%p) {\n", data);
+  fprintf (fp, "data_size = [%d]  data = [%.*s]\n", QFILE_GET_TUPLE_LENGTH (data2->tuple),
+	   QFILE_GET_TUPLE_LENGTH (data2->tuple), data2->tuple);
+
+  fprintf (fp, "key : ");
+  for (int i = 0; i < key2->val_count; i++)
+    {
+      db_fprint_value (fp, key2->values[i]);
+      fprintf (fp, " ");
+    }
+  fprintf (fp, "\n}");
+
+  return true;
+}
+
+/*
+ * qdata_copy_hscan_key () - deep copy aggregate key
+ *   returns: pointer to new aggregate hash key
+ *   thread_p(in): thread
+ *   key(in): source key
+ */
+HASH_SCAN_KEY *
+qdata_copy_hscan_key (cubthread::entry * thread_p, HASH_SCAN_KEY * key, REGU_VARIABLE_LIST probe_regu_list,
+		      val_descr * vd)
+{
+  HASH_SCAN_KEY *new_key = NULL;
+  int i = 0;
+  DB_TYPE vtype1, vtype2;
+  TP_DOMAIN_STATUS status = DOMAIN_COMPATIBLE;
+
+  if (key)
+    {
+      /* make a copy */
+      new_key = qdata_alloc_hscan_key (thread_p, key->val_count, false);
+    }
+
+  if (new_key)
+    {
+      /* copy values */
+      new_key->val_count = key->val_count;
+      for (i = 0; i < key->val_count; i++)
+	{
+	  vtype1 = REGU_VARIABLE_GET_TYPE (&probe_regu_list->value);
+	  vtype2 = DB_VALUE_DOMAIN_TYPE (key->values[i]);
+
+	  if (vtype1 != vtype2)
+	    {
+	      new_key->values[i] = pr_make_value ();
+	      if (new_key->values[i] == NULL)
+		{
+		  new_key->free_values = true;
+		  qdata_free_hscan_key (thread_p, new_key, i);
+		  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, sizeof (DB_VALUE *));
+		  return NULL;
+		}
+
+	      status = tp_value_coerce (key->values[i], new_key->values[i], probe_regu_list->value.domain);
+	      if (status != DOMAIN_COMPATIBLE)
+		{
+		  new_key->free_values = true;
+		  qdata_free_hscan_key (thread_p, new_key, ++i);
+		  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_TP_CANT_COERCE, 2, pr_type_name (vtype2),
+			  pr_type_name (vtype1));
+		  return NULL;
+		}
+	    }
+	  else
+	    {
+	      new_key->values[i] = pr_copy_value (key->values[i]);
```

Comment:
we need to check for pr_copy_value (key->values[i]) returns null in case of out of memory.
please refer at object_primitive.c::pr_copy_value()  a comment related to this situation.

Reply:
I fixed it